src/DocumentManager.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2021, Kung-Yao Hsieh, Shang-Chen LIN, Yi-Hang Xie. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef HTTPLIB_SRC_DOCUMENTMANAGER_H
+#define HTTPLIB_SRC_DOCUMENTMANAGER_H
+
+#include <string>
+#include <vector>
+
+#include "DocumentSession.h"
+
+class DocumentSession;
+
+/** @brief Store document modify revision */
+typedef struct{
+    /** @brief Who made this revision. */
+    std::string user_id;
+    /** @brief Relative insert position from document start and end. */
+    int start_position;
+    int end_position;
+    /** @brief insertData and deleteLength will only exist one at the same time. */
```

Comment:
How do you tell if we look at `insert_data` or `delete_length`?

Reply:
added is_insert to distinguish insert or delete revision